heron/tmaster/src/cpp/manager/tmaster.cpp

```diff
@@ -142,6 +142,14 @@ void TMaster::EstablishTMaster(EventLoop::Status) {
   auto cb = [this](proto::system::StatusCode code) { this->SetTMasterLocationDone(code); };
 
   state_mgr_->SetTMasterLocation(*tmaster_location_, std::move(cb));
+
+  // if zk lost the tmaster location, tmaster quits to bail out and re-establish its location
+  auto cb2 = [this]() {
+    LOG(ERROR) << " lost tmaster location in zk state manager. Bailing out..." << std::endl;
```

Comment:
1. IMO, Fail fast in this case is better. TMaster only knows the fact that its location is lost but doesn't know the exact cause. So cleaning itself up and then resetting its location from scratch is cleaner.
2. From implementation side, "set the location again" is not that convenient to implement. It requires a lot of changes in other parts(stmgr, tmaster).